src/MongoDB.Driver.Core/Core/Configuration/ConnectionSettings.cs

```diff
@@ -118,7 +125,40 @@ public TimeSpan MaxLifeTime
             get { return _maxLifeTime; }
         }
 
-        // methods
+        // internal methods
+        internal ConnectionSettings Fork()
```

Comment:
the main change in this PR. This method creates a new instance of ConnectionSettings and calls the _authenticatorsConfigurator delegate to initialize a new instance for authenticators if any